parquet/src/main/java/org/apache/iceberg/parquet/PruneColumns.java

```diff
@@ -162,9 +168,11 @@ private Integer getId(Type type) {
     return type.getId() == null ? null : type.getId().intValue();
   }
 
-  private boolean isStruct(Type field) {
+  private boolean isStruct(Type field, NestedField expected) {
     if (field.isPrimitive()) {
       return false;
+    } else if (expected.type() == org.apache.iceberg.types.Types.VariantType.get()) {
```

Comment:
These changes were needed to project the sub-fields of a variant because it looks like a struct.

Reply:
Here's another case where I feel like the `Type:isVariantType` would make things more clear.

Reply:
Yes, I'll update this when the other PR from @aihuaxu is in.